resources/io/jenkins/infra/docker/pod-template.yml

```diff
@@ -0,0 +1,51 @@
+---
+apiVersion: "v1"
+kind: "Pod"
+metadata:
+  labels:
+    jenkins: "agent"
+  annotations:
+    container.apparmor.security.beta.kubernetes.io/builder: unconfined
+    container.seccomp.security.alpha.kubernetes.io/builder: unconfined
+spec:
+  tolerations:
+  - key: "os"
+    operator: "Equal"
+    value: "linux"
+    effect: "NoSchedule"
+  - key: "profile"
+    operator: "Equal"
+    value: "highmem"
+    effect: "NoSchedule"
+  affinity:
+    nodeAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+        nodeSelectorTerms:
+        - matchExpressions:
+          - key: kubernetes.io/os
+            operator: In
+            values:
+            - linux
+      preferredDuringSchedulingIgnoredDuringExecution:
+      - weight: 1
+        preference:
+          matchExpressions:
+          - key: agentpool
+            operator: In
+            values:
+            - highmemlinux
```

Comment:
Do we really need a highmemlinux instance? My guess is no

Reply:
Hey, at least the PR's check was fast... :D

Reply:
Removed the `affinity` block as I prefer delegating to the orchestrator as much as possible